common/blobstore/metricClient.go

```diff
@@ -68,6 +68,19 @@ func (c *metricClient) Download(ctx context.Context, bucket string, key blob.Key
 	return resp, err
 }
 
+func (c *metricClient) GetTags(ctx context.Context, bucket string, key blob.Key) (map[string]string, error) {
+	c.metricsClient.IncCounter(metrics.BlobstoreClientGetTagsScope, metrics.CadenceClientRequests)
+
+	sw := c.metricsClient.StartTimer(metrics.BlobstoreClientGetTagsScope, metrics.CadenceClientLatency)
+	resp, err := c.client.GetTags(ctx, bucket, key)
+	sw.Stop()
+
+	if err != nil {
+		c.metricsClient.IncCounter(metrics.BlobstoreClientGetTagsScope, metrics.CadenceClientFailures)
```

Comment:
General pattern is to report errors which count towards availability of that API.  User errors should not be counted as errors like "BlobNotExist".

Reply:
I based this off the other metrics clients which follow a pattern like this

```
func (c *metricClient) DescribeDomain(
	ctx context.Context,
	request *shared.DescribeDomainRequest,
	opts ...yarpc.CallOption,
) (*shared.DescribeDomainResponse, error) {

	c.metricsClient.IncCounter(metrics.FrontendClientDescribeDomainScope, metrics.CadenceClientRequests)

	sw := c.metricsClient.StartTimer(metrics.FrontendClientDescribeDomainScope, metrics.CadenceClientLatency)
	resp, err := c.client.DescribeDomain(ctx, request, opts...)
	sw.Stop()

	if err != nil {
		c.metricsClient.IncCounter(metrics.FrontendClientDescribeDomainScope, metrics.CadenceClientFailures)
	}
	return resp, err
}
```

In this case DescribeDomain could return error entity not exists but we still emit a metric based on this error.

I agree with you that these errors should not count against API availability but I think changing this involves changing how we do all our clients?